src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    private static final class AnnotationManager {
+
+        private final List<String> annotationGroupsToUse;
+        private final List<String> annotationsToUse;
+        private final List<String> annotationsToExclude;
+
+        private AnnotationManager(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            this.annotationGroupsToUse = annotationGroupsToUse;
+            this.annotationsToUse = annotationsToUse;
+            this.annotationsToExclude = annotationsToExclude;
+
+            final Set<String> allAnnotationNames = new HashSet<>(AnnotationManager.getAllAnnotationNames());
+            final Set<String> unknownAnnots = Sets.difference(new HashSet<>(annotationsToUse), allAnnotationNames);
+            if (!unknownAnnots.isEmpty()){
+                throw new BadArgumentValue("annotation", "Annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownAnnotsExclude = Sets.difference(new HashSet<>(annotationsToExclude), allAnnotationNames);
+            if (!unknownAnnotsExclude.isEmpty()){
+                throw new BadArgumentValue("annotation", "Excluded annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownGroups =  Sets.difference(new HashSet<>(annotationGroupsToUse), new HashSet<>(AnnotationManager.getAllAnnotationGroupNames()));
+            if (!unknownGroups.isEmpty()){
+                throw new BadArgumentValue("group", "Unknown annotation group in " + unknownGroups + ". Known groups are " + knownSubInterfaceSimpleNames(AnnotationKind.class));
+            }
+        }
+
+        private List<String> knownSubInterfaceSimpleNames(Class<AnnotationKind> annotationKindClass) {
+            return knownSubInterfaces(annotationKindClass).stream().map(c ->c.getSimpleName()).collect(Collectors.toList());
+        }
+
+        private Set<Class<?>> knownSubInterfaces(final Class<?> clazz) {
+            final ClassFinder finder = new ClassFinder();
+            finder.find(clazz.getPackage().getName(), clazz);
+            return finder.getClasses().stream().filter(cl -> cl.isInterface()).collect(Collectors.toSet());
+        }
+
+        /**
+         * An annotation will be included only when:
+         * - it is in one of the annotation groups or
+         * - it is listed explicitly
+         * - and it is not excluded explicitly.
+         */
+        static AnnotationManager ofSelectedMinusExcluded(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            final List<String> groups = new ArrayList<>(annotationGroupsToUse);//make copy
+            final List<String> annots = new ArrayList<>(annotationsToUse);//make copy
```

Comment:
that;s because it's copied later on. I'll make a copy here too for symmetry. This does not matter for performance because it happens once